### PR TITLE
Add feature to change archived status for dropbox

### PR DESCRIPTION
**Pull request by kosiakkatrina**
**What**
Add dropbox status and a button allowing to set archived/unarchived status of the dropbox.
image
image

**Why**
To allow better dropbox management.

### CODE REVIEW

File: lambda.js [hunk 1]
```diff
@@ -128,6 +129,18 @@ api.get('/dropboxes/:id', async (req, res) => {
   );
 });
 
+api.post('/dropboxes/:id/archive', async (req, res) => {
+  const { archiveStatus } = req.body;
+  const dropboxId = req.params.id;
+  console.log(archiveStatus, dropboxId);
```
Comment (t-davies) [inline lambda.js:135]: Can probably remove this now?

File: lib/use-cases/updateArchiveStatus.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module.exports = ({ gateways: { dropbox: dropboxes } }) => {
```
Comment (t-davies) [inline lib/use-cases/updateArchiveStatus.js:1]: Maybe rename this file to match the others - so `UpdateArchiveStatus` vs `updateArchiveStatus`

File: templates/readonlyDropbox.hbs [hunk 3]
```diff
@@ -29,7 +29,40 @@
...
+            method: 'POST',
+            body: JSON.stringify({ archiveStatus }),
+            mode: 'no-cors'
+          }).then(function () {
+            window.location.reload();
+          })
+        }
+
+        document.querySelector('#archive') ? document.querySelector('#archive').addEventListener('click', function (event) { doArchive(event, true); }) : null;
+        document.querySelector('#unarchive') ? document.querySelector('#unarchive').addEventListener('click', function (event) { doArchive(event, false); }) : null;
```
Comment (t-davies) [inline templates/readonlyDropbox.hbs:65]: Wonder if we could refactor this into something like...
```
function tryAddEventListener(selector, event, handler) {
  const element = document.querySelector(selector);
  if (element) {
    element.addEventListener(event, handler);
  }
}
```

Comment (t-davies) [reply templates/readonlyDropbox.hbs:65]: And then we can swap this to make it less repetitive?
```
tryAddEventListener('#archive', 'click', function() { ...etc });
tryAddEventListener('#unarchive', 'click', function() { ...etc });
```

Comment (t-davies) [reply templates/readonlyDropbox.hbs:65]: But I don't think it's essential, just nice-to-have.

Comment (mmmmillar) [reply templates/readonlyDropbox.hbs:65]: Yep, would be nice to tidy that up a bit

Comment (kosiakkatrina) [reply templates/readonlyDropbox.hbs]: Sounds good!

**Pull request merged**